Update demos to TinyMCE 8

MitchC1999 opened the pull request:
Update demos to TinyMCE 8

MitchC1999 committed:
TINY: Update demos to TinyMCE 8

MitchC1999 committed:
Fix slash commands demo

ShiridiGandham reviewed (approved):
LGTM

lostkeys commented on templates-and-layout/locked-down-template-inline-bordered.html:
A bug introduced between 5 and 8: Put the cursor in editable element and type something, move it to title and observe editor appears disabled but the toolbar still functions (eg try pressing press bold)

lostkeys commented on tinymce-basics/useful_styles_for_content_css.html:
```diff
@@ -46,8 +46,8 @@
                  */
 
-                /* Make our invisible overlay be positioned relative to the hr */
-                hr {
```
Result in invalid CSS

lostkeys commented on tinymce-basics/useful_styles_for_content_css.html:
```diff
@@ -124,7 +124,7 @@ <h1>Useful content css</h1>
         <p>The default content css is very basic and unopinionated to allow you as a developer to integrate the styles you want in order to get a good WYSIWYG writing environment.</p>
         <p>That said, here are some tecniques we often implement ourselves that improves the editing experience in TinyMCE</p>
         <p><em>Inspect the source code to see the actual css used</em></p>
-        <h2>Improve hr clickability</h2>
```
The purpose of this sentence is to show how to improve the clickability of the `hr` element, and removing the word "hr" makes the sentence not make any sense

lostkeys commented on tokens/text-based-token.html:
Remove this demo. TinyMCE now have a premium merge tags plugin and we should not teach customers how to create their own

lostkeys committed:
* Updated license

tinymce-basics/useful_styles_for_content_css.html
* Fixed invalid css
* Re-added "hr" in the content
* Removed `noneditable` plugin
* Updated all documentation links

tinymce-basics/badges.html
* Removed `noneditable` plugin
* Updated all documentation links

tinymce-basics/custom-toolbar-button-menu-and-context-toolbar.html
* Removed `noneditable` plugin
* Updated all documentation links

tinymce-basics/webapp-description-comment-reply-editor.html
* Correctly use text pattern feature
* Removed text pattern plugin
* Updated all documentation links
* Added markdown plugin

tinymce-basics/bootstrap-rich-text-form.html
* Updated all documentation links

tinymce-basics/right-aligned-toolbar-button.html
* Removed demo because we now have various badges showing up here

tinymce-basics/format-link-as-button.html
* Updated all documentation links

tinymce-basics/fonts.html
* Updated all documentation links
* replaced font_formats with font_family_formats to make custom fonts show up

images/toggle-image-size-context-toolbar.html
* Updated all documentation links

images/image-alt-text-context-form.html
* Updated all documentation links

tables/customizing-the-table-toolbar.html
* Updated all documentation links
* TODO: Commented out the cut/copy/paste row & column buttons preventing the menu from working, waiting for someone more talented to fix them using the new fake clipboard api: https://www.tiny.cloud/docs/tinymce/latest/apis/tinymce.fakeclipboard/

tables/table-row-templates.html
* TODO: someone needs to update the copy paste row using the new fake clipboard api: https://www.tiny.cloud/docs/tinymce/latest/apis/tinymce.fakeclipboard/ otherwise the whole premise of the demo is gone
* Updated all documentation links

templates-and-layout/advanced-placeholder.html
* Removed demo because the new enhanced templates plugin does not have the same features as the template plugin it replaced, and cannot be used with this demo, therefore the premise of the demo is lost.

templates-and-layout/multiple-columns-using-css-columns.html
* Removed demo because using css this way is not something TinyMCE supports very well, and I feel it is unwise to lead someone down this route.

templates-and-layout/locked-down-template-inline-default.html
* Updated all documentation links

templates-and-layout/locked-down-template-inline-context-toolbars.html
* Updated all documentation links

templates-and-layout/locked-down-template-inline-context-toolbars.html
* Updated all documentation links

templates-and-layout/locked-down-template-inline-bordered.html
* Updated all documentation links
* Re-added the borderless skin to fix broken toolbar visuals

email/webmail-rich-text-editor.html
* Switched to naked skin because the outside skin had the editor outline on top of the bottom toolbar
* Updated all documentation links

email/marketing-email-inline-mode.html
* Updated all documentation links

email/marketing-email-inline-mode-fixed-toolbar.html
* Updated all documentation links

email/marketing-email.html
* Updated the demo to use the mergetags plugin
* Updated all documentation links

cool-demos/expand-editor-and-show-toolbar.html
* Fixed invalid CSS
* Updated all documentation links
* Note that toolbar bottom with the new tooltips makes it almost impossible to use the toolbar

cool-demos/conditional-blocks.html
* Updated all documentation links
* Removed `noneditable` plugin

cool-demos/slash-commands.html
* Added the `tableofcontents` plugin
* Updated all documentation links

lostkeys commented on email/webmail-rich-text-editor.html:
Had to change the skin

metricjs commented on cool-demos/slash-commands.html:
```diff
@@ -118,14 +119,14 @@
                 // configure the autocompleter to display a menu opposed a row of icons
-                // https://www.tiny.cloud/docs/ui-components/autocompleter/#configurationoptions
-                ch: '/',
+                // https://www.tiny.cloud/docs/tinymce/latest/autocompleter/
```
```suggestion
```
Duplicate of the link a couple lines above, so unnecessary?

metricjs commented on cool-demos/expand-editor-and-show-toolbar.html:
```diff
@@ -1,7 +1,7 @@
 <!--
     This demo showcases a minimalist editor that expands on focus
```
In browser, this gets mashed into one line of comment so the lack of a full stop is a little confusing. Hence the nitpick

metricjs commented on email/marketing-email.html:
```diff
@@ -1,17 +1,17 @@
+to the rich text parts. All the table layout are outside
+of TinyMCE. This makes it easy for you to to add other
+logic such as inserting blocks and drag and drop reordering
+yourself.
```
```suggestion
The proper way to do marketing emails with TinyMCE is to use
inline mode, and move all table layouts outside of TinyMCE.
Tables in TinyMCE are first and foremost to display tabular
data, not for layouts. Doing table layouts in TinyMCE,
while possible, is not recommended. It is simply too easy
to accidentally break the design.

Here is an example of how to do a marketing email using
TinyMCE inline mode. The TinyMCE editor is only applied
to the rich text parts. The table layout is outside
of TinyMCE. This makes it easy for you to to add other
logic such as inserting blocks and drag and drop reordering
yourself.
```
nitpicking the wording

metricjs commented on tables/customizing-the-table-toolbar.html:
```diff
@@ -32,26 +30,26 @@
             menubar: false,
 
             // Tip! To make TinyMCE leaner, only include the plugins you actually need.
             plugins: 'link lists image table code',
```
```suggestion
            plugins: 'link lists image table code emoticons',
```
emoticons is in the toolbar config but missing from here

metricjs commented on tables/customizing-the-table-toolbar.html:
```diff
@@ -109,10 +107,10 @@
-                // TinyMCE 5.4 introduced improved APIs for inserting tables.
+                // TinyMCE has APIs for inserting tables.
                 // Here's a toolbar button that displays a table grid picker
                 // The table picker is unfortunately undocumented at this stage
```
Feels weird to just say this. Is this something we _want_ to document? Looks like we don't document fancy menu items at all. "At this stage" implies that we might document it down the track...

metricjs replied:
Alternatively - why do we have an "inserttable" menu item but not a toolbar button in TinyMCE? Should we add one?? Questions for outside the scope of this PR.

TheSpyder replied:
It's a question that comes up occasionally from customers as well. I sometimes regret my inability to put us on a good architecture with TinyMCE 5 that would have made both work automatically.

metricjs replied:
Probably worth documenting it at least, if customers are asking. I'll flag it with Karl

metricjs commented on tables/customizing-the-table-toolbar.html:
```diff
@@ -384,84 +386,84 @@
                 tinymce.activeEditor.execCommand('mceTablePasteRowAfter');
             },
             headerColumn: () => {
                 // Check the type of the selected table column and inverse it
```
```suggestion
                // Check the type of the selected table column and invert it
```

metricjs commented on tables/table-row-templates.html:
Would need a dev to advise on if this functionality is even possible with TinyMCE 8 for sure. I can't find a way to do it after looking at the table plugin's commands and APIs for a bit.

lostkeys committed:
Updated spelling, wording and fixed minor errors

lostkeys commented on email/webmail-rich-text-editor.html:
```diff
@@ -73,16 +73,16 @@
             // using the block_formats option.
-            // https://www.tiny.cloud/docs/configure/editor-appearance/#block_formats
+            // https://www.tiny.cloud/docs/tinymce/latest/user-formatting-options/#block_formats
             block_formats: 'Normal=p; Heading=h1; Sub heading=h2; Small=small',
```
Removed block formats, don't know why I did it that way

lostkeys committed:
Fixed clipboard for table-row-templates

lostkeys committed:
Removed copy/paste rows and columns because the low-level API replacing the old table clipboard apis are not for public consumption
